src/ngraph/frontend/onnx_import/core/tensor.hpp

```diff
@@ -130,14 +153,14 @@ namespace ngraph
                 {
                     if (tensor.has_raw_data())
                     {
-                        return detail::__get_raw_data<double>(tensor.raw_data());
+                        return detail::__get_raw_data<double>(tensor.raw_data(),
+                                                              tensor.data_type());
                     }
                     if (tensor.data_type() == onnx::TensorProto_DataType_DOUBLE)
                     {
                         return detail::__get_data<double>(tensor.double_data());
                     }
-                    if ((tensor.data_type() == onnx::TensorProto_DataType_FLOAT) ||
-                        (tensor.data_type() == onnx::TensorProto_DataType_FLOAT16))
```

Comment:
If we would get a tensor with `FLOAT16` data type which is not stored in raw_data field we will get an exception.
You should probably move `(tensor.data_type() == onnx::TensorProto_DataType_FLOAT16))` check to INT32 section instead of just removing it.
ONNX doc:
```
// For int32, uint8, int8, uint16, int16, bool, and float16 values
// float16 values must be bit-wise converted to an uint16_t prior
// to writing to the buffer.
// When this field is present, the data_type field MUST be
// INT32, INT16, INT8, UINT16, UINT8, BOOL, or FLOAT16
repeated int32 int32_data = 5 [packed = true];
```
@postrational what do you think about it?

Reply:
Ok, I could do that but I don't have a way to test that case. Could someone provide one? Otherwise, maybe it's better to get an exception and implement it once we can test it.

Reply:
Ok, we can defer it for now I think.